From a00ce68f8b08abb925bf22c0c46858958dd93866 Mon Sep 17 00:00:00 2001
Message-Id: <a00ce68f8b08abb925bf22c0c46858958dd93866.1616668703.git.stefan@agner.ch>
From: Stefan Agner <stefan@agner.ch>
Date: Thu, 25 Mar 2021 11:35:12 +0100
Subject: [PATCH] Allow to explicitly disable libsystemd-login

Currently libsystemd-login is used to inhibit poweroff, sleep and
reboot. The library makes calls to systemd-logind which in turn needs
quite some services to be present. This can be problematic when using
udisks in early boot stages.

Signed-off-by: Stefan Agner <stefan@agner.ch>
---
 configure.ac | 21 +++++++++++++--------
 1 file changed, 13 insertions(+), 8 deletions(-)

diff --git a/configure.ac b/configure.ac
index 53917998..0b3f0c9a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -197,16 +197,21 @@ if test "x$enable_daemon" = "xyes"; then
                 [AC_DEFINE([UUID_STR_LEN], 37, [compat UUID_STR_LEN define])],
                 [[#include <uuid/uuid.h>]])
 
-  PKG_CHECK_MODULES(LIBSYSTEMD_LOGIN, [libsystemd >= 209], [have_libsystemd_login=yes],
-                    [PKG_CHECK_MODULES(LIBSYSTEMD_LOGIN, [libsystemd-login >= 44 libsystemd-daemon],
-                    [have_libsystemd_login=yes],
-                    [have_libsystemd_login=no])])
-  if test "x$have_libsystemd_login" = "xyes"; then
-    AC_DEFINE([HAVE_LIBSYSTEMD_LOGIN], 1, [Define to 1 if libsystemd-login is available])
+  have_libsystemd_login=no
+  AC_ARG_ENABLE(libsystemd-login, AS_HELP_STRING([--disable-libsystemd-login], [disable libsystemd-login support]))
+  if test "x$enable_libsystemd_login" != "xno"; then
+    PKG_CHECK_MODULES(LIBSYSTEMD_LOGIN, [libsystemd >= 209], [have_libsystemd_login=yes],
+                      [PKG_CHECK_MODULES(LIBSYSTEMD_LOGIN, [libsystemd-login >= 44 libsystemd-daemon],
+                      [have_libsystemd_login=yes],
+                      [have_libsystemd_login=no])])
+    if test "x$have_libsystemd_login" = "xyes"; then
+      AC_DEFINE([HAVE_LIBSYSTEMD_LOGIN], 1, [Define to 1 if libsystemd-login is available])
+    fi
+    AC_SUBST(LIBSYSTEMD_LOGIN_CFLAGS)
+    AC_SUBST(LIBSYSTEMD_LOGIN_LIBS)
   fi
+  AM_CONDITIONAL([HAVE_LIBSYSTEMD_LOGIN], [test "x$have_libsystemd_login" = "xyes"])
   AC_SUBST(HAVE_LIBSYSTEMD_LOGIN)
-  AC_SUBST(LIBSYSTEMD_LOGIN_CFLAGS)
-  AC_SUBST(LIBSYSTEMD_LOGIN_LIBS)
 
   PKG_CHECK_MODULES(LIBELOGIND, [libelogind >= 219],
                     [have_libelogind=yes],
-- 
2.30.2

